packages/cli-platform-ios/src/tools/prompts.ts

```diff
@@ -35,14 +35,27 @@ export async function promptForConfigurationSelection(
 }
 
 export async function promptForDeviceSelection(
-  availableDevices: Device[],
+  devices: Device[],
+  lastUsedIOSDeviceId?: string,
 ): Promise<Device | undefined> {
+  const sortedDevices = devices;
```

Comment:
`sortedDevices` still references `devices` and is then mutated. You'd need to copy the `devices` array to avoid mutating it with `unshift` on line 49